internal/config/instance.go

```diff
@@ -113,65 +118,86 @@ func Get() (*Instance, error) {
 }
 
 // Set sets a value at the given key
-func (i *Instance) Set(key string, value interface{}) {
+func (i *Instance) Set(key string, value interface{}) error {
 	i.rwLock.Lock()
 	defer i.rwLock.Unlock()
-	i.viper.Set(key, value)
+
+	configData, err := i.configFileData()
+	if err != nil {
+		return err
+	}
+
+	configData[strings.ToLower(key)] = value
+	i.data = configData
+
+	err = i.save()
+	if err != nil {
+		return err
+	}
+
+	return nil
 }
 
-// GetString retrieves a string for a given key
-func (i *Instance) GetString(key string) string {
+func retryLock(pl *lockfile.PidLock, attempts int, interval time.Duration) error {
+	lockedErr := &lockfile.AlreadyLockedError{}
+	_, err := pl.TryLock()
+	if err != nil {
+		if !errors.As(err, &lockedErr) {
+			return errs.Wrap(err, "Unexpected error attempting to acquire lock file")
+		}
+		if attempts--; attempts > 0 {
+			time.Sleep(interval)
+			return retryLock(pl, attempts, interval)
+		}
+		return err
+	}
+
+	return nil
+}
```

Comment:
I think this should be part of our pidlock mechanic. Maybe a `WaitForLock(timeout)` function.